packages/protocol/contracts/dapis/SamplePriceDataFeed.sol

```diff
@@ -2,15 +2,33 @@
 pragma solidity 0.8.4;
 
 import "./RrpDapiServer.sol";
+import "./Api3Adminship.sol";
 
 // A sample price data feed implemented as a dAPI client
-contract SamplePriceDataFeed {
+contract SamplePriceDataFeed is Api3Adminship {
   RrpDapiServer public rrpDapiServer;
+
   bytes16 public latestDapiId;
   int256 public latestAnswer;
 
-  constructor(address _rrpDapiServer) {
+  uint256 public cooldownTime = 1 days; // TODO: is this the right value?
+  mapping(address => uint64) public adminToDapiLastUpdated;
+
+  /// @dev Reverts if the caller tries to update a dAPI again before the cooldown period has elapsed
+  modifier cooldownTimeEnded() {
+    require(
+      adminToDapiLastUpdated[msg.sender] == 0 || // first time updating will not have a last updated value
+        block.timestamp >= (adminToDapiLastUpdated[msg.sender] + cooldownTime),
+      "Cooldown period has not finished"
+    );
+    _;
+  }
+
+  constructor(address _rrpDapiServer, address _metaAdmin) Api3Adminship(_metaAdmin) {
+    require(_rrpDapiServer != address(0), ERROR_ZERO_ADDRESS);
     rrpDapiServer = RrpDapiServer(_rrpDapiServer);
+    // TODO: should we set superadmin status to technical team deploying the contract?
```

Comment:
You mean, don't supply `_metaAdmin` and just set to the address of the deployer? Yea, I agree with you, I like that better.

Reply:
True, but it also makes sense that the team would also deploy the contract. @bbenligiray thoughts?

Reply:
I think a dev would deploy the contract using their personal wallet, setting themselves as the meta admin. Then they would give the multisig adminship, then they would transfer the meta adminship to the DAO. Then it's verified that it's set up correctly and then it starts being used.